test/instrumentation/modules/hapi.js

```diff
@@ -15,6 +16,28 @@ var test = require('tape')
 var http = require('http')
 var Hapi = require('hapi')
 
+var originalCaptureError = agent.captureError
+
+test('extract URL from request', function (t) {
+  resetAgent(function (endpoint, headers, data, cb) {
+    var request = data.errors[0].context.request
+    t.equal(request.method, 'GET')
+    t.equal(request.url.pathname, '/captureError')
+    t.equal(request.url.search, '?foo=bar')
+    t.equal(request.url.raw, '/captureError?foo=bar')
+    t.equal(request.url.hostname, 'localhost:' + server.info.port)
+    t.equal(request.socket.encrypted, false)
+    server.stop()
+    t.end()
+  })
+
+  agent.captureError = originalCaptureError
```

Comment:
Should this set the `captureError` to the original value _after_ running the specified assertions in `resetAgent`?

Reply:
No, `resetAgent` will add a mock `captureError` function, which we need in all the other tests. But in this particular test we actually want the real thing. So after running `resetAgent`, but before running the assertions, we just restore the `captureError` function to its original value.

Reply:
ah got it, thx for the explanation.